substrate/runtime/timestamp/src/lib.rs

```diff
@@ -35,54 +35,69 @@ extern crate serde_derive;
 extern crate substrate_primitives;
 extern crate substrate_runtime_primitives as runtime_primitives;
 extern crate substrate_runtime_system as system;
+extern crate substrate_runtime_consensus as consensus;
 extern crate substrate_codec as codec;
 
 use runtime_support::{StorageValue, Parameter};
 use runtime_support::dispatch::Result;
-use runtime_primitives::traits::{HasPublicAux, Executable, MaybeEmpty};
+use runtime_primitives::traits::{Executable, MaybeEmpty, SimpleArithmetic, As, Zero};
 
-pub trait Trait: HasPublicAux + system::Trait {
+pub trait Trait: consensus::Trait where
+	<Self as consensus::Trait>::PublicAux: MaybeEmpty
+{
 	// the position of the required timestamp-set extrinsic.
-	const SET_POSITION: u32;
+	const TIMESTAMP_SET_POSITION: u32;
 
-	type Value: Parameter + Default;
+	type Moment: Parameter + Default + SimpleArithmetic + As<Self::BlockNumber>;
```

Comment:
`Moment: As<Self::BlockNumber>` could be confusing (if not getting into usage details) - is it simply `transmute(moment) -> BlockNumber` or is it some complex mapping of `timestamp => best block number at this timestamp`. Maybe worth a comment? Or probably it is just me :)